cairo/src/image_surface.rs

```diff
@@ -187,6 +192,9 @@ impl AsRef<[u8]> for ImageSurfaceDataOwned {
         unsafe {
             let ptr = ffi::cairo_image_surface_get_data(self.surface.to_raw_none());
             debug_assert!(!ptr.is_null());
+            if len == 0 {
```

Comment:
Since it's only a `debug_assert`, shouldn't you check for `ptr.is_null()` in any case?